### ISSUE TITLE
There was an error in the forked process null

### ISSUE THREAD

**Issue by kashifalam** [cbeust/testng#1962]
### TestNG Version
> 6.9.10
### Expected behavior
Test should start running when run through jenkins  on server

### Actual behavior
Getting error while running through jenkins
Able to run on local device by running testng.xml file directly

```
[INFO] Tests run: 0, Failures: 0, Errors: 0, Skipped: 0
[INFO] 
[ERROR] There are test failures.

Please refer to /var/lib/jenkins/workspace/ODIN Automation/target/surefire-reports for the individual test results.
Please refer to dump files (if any exist) [date].dump, [date]-jvmRun[N].dump and [date].dumpstream.
There was an error in the forked process
null
org.apache.maven.surefire.booter.SurefireBooterForkException: There was an error in the forked process
null
	at org.apache.maven.plugin.surefire.booterclient.ForkStarter.fork(ForkStarter.java:657)
	at org.apache.maven.plugin.surefire.booterclient.ForkStarter.run(ForkStarter.java:283)
	at org.apache.maven.plugin.surefire.booterclient.ForkStarter.run(ForkStarter.java:246)
	at org.apache.maven.plugin.surefire.AbstractSurefireMojo.executeProvider(AbstractSurefireMojo.java:1149)
```

### Is the issue reproductible on runner?
- [x] Maven

**LOG FILE**
```
[INFO] Running TestSuite
Starting ChromeDriver 2.41.578700 (2f1ed5f9343c13f73144538f15c00b370eda6706) on port 18653
Only local connections are allowed.
1484531981
<p id ="infoMessage"> creating new webdriver instance </p>
<p id ="infoMessage"> Test execution starts </p>
1484531981
<p id ="infoMessage"> Test execution starts </p>
1484531981
<p id ="infoMessage"> Test execution starts </p>
1484531981
<p id ="infoMessage"> Test execution starts </p>
1484531981
<p id ="infoMessage"> Test execution starts </p>
1484531981
<p id ="infoMessage"> Test execution starts </p>
1484531981
<p id ="infoMessage"> Test execution starts </p>
1484531981
<p id ="infoMessage"> Test execution starts </p>
1484531981
<p id ="infoMessage"> Test execution starts </p>
<p id ="infoMessage"> creating new webdriver instance </p>
[INFO] 
[INFO] Results:
[INFO] 
[INFO] Tests run: 0, Failures: 0, Errors: 0, Skipped: 0
[INFO] 
[ERROR] There are test failures.

Please refer to /var/lib/jenkins/workspace/ODIN Automation/target/surefire-reports for the individual test results.
Please refer to dump files (if any exist) [date].dump, [date]-jvmRun[N].dump and [date].dumpstream.
There was an error in the forked process
null
org.apache.maven.surefire.booter.SurefireBooterForkException: There was an error in the forked process
null
	at org.apache.maven.plugin.surefire.booterclient.ForkStarter.fork(ForkStarter.java:657)
	at org.apache.maven.plugin.surefire.booterclient.ForkStarter.run(ForkStarter.java:283)
	at org.apache.maven.plugin.surefire.booterclient.ForkStarter.run(ForkStarter.java:246)
	at org.apache.maven.plugin.surefire.AbstractSurefireMojo.executeProvider(AbstractSurefireMojo.java:1149)
	at org.apache.maven.plugin.surefire.AbstractSurefireMojo.executeAfterPreconditionsChecked(AbstractSurefireMojo.java:991)
	at org.apache.maven.plugin.surefire.AbstractSurefireMojo.execute(AbstractSurefireMojo.java:837)
	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:134)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:207)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:153)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:145)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:116)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:80)
	at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:51)
	at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:128)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:307)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:193)
	at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:106)
	at org.jvnet.hudson.maven3.launcher.Maven33Launcher.main(Maven33Launcher.java:129)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launchStandard(Launcher.java:330)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:238)
	at jenkins.maven3.agent.Maven33Main.launch(Maven33Main.java:176)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at hudson.maven.Maven3Builder.call(Maven3Builder.java:139)
	at hudson.maven.Maven3Builder.call(Maven3Builder.java:70)
	at hudson.remoting.UserRequest.perform(UserRequest.java:210)
	at hudson.remoting.UserRequest.perform(UserRequest.java:53)
	at hudson.remoting.Request$2.run(Request.java:364)
	at hudson.remoting.InterceptingExecutorService$1.call(InterceptingExecutorService.java:72)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)

[JENKINS] Recording test results
[INFO] 
[INFO] --- maven-jar-plugin:2.4:jar (default-jar) @ Chitti-ODIN ---
[INFO] Building jar: /var/lib/jenkins/workspace/ODIN Automation/target/Chitti-ODIN-0.0.3.jar
[INFO] 
```

**TESTNG FILE**
```xml
<?xml version="1.0" encoding="UTF-8"?><!DOCTYPE suite SYSTEM "http://testng.org/testng-1.0.dtd" >
<suite name="ODIN Tests" verbose="1">
    <!-- verbose="1"  -->
    <listeners>
        <listener class-name="common.helpers.TestNGHtmlReportGenerator"/>
        <listener class-name="common.helpers.TestNGTestListener"/>
    </listeners>
    <parameter name="retry_enabled" value="false"/>
    <parameter name="enable_logs" value="true"/>
    <parameter name="webdriver_timeout" value="10000"/>


    <parameter name="environment" value="PREPROD"/>
    <parameter name="browser" value="CHROME"/>
    <!--<parameter name="ODIN_URL" value="https://odin-qa.ailiens.com/products" />	 https://www.nnnow.com/ -->
    <!-- https://odin-qa.ailiens.com/  https://mumbai-prod.nnnow.com/-->

    <!-- _linux64 -->
    <!-- /Users/tech/Downloads/chromedriver3 -->

    <!-- <test name="ODIN_SignUP_Loginv1">
        <groups>
            <run>
                <include name="Regress_Odin" />
            </run>
        </groups>
        <packages>
            <package name="com.odin.Chitti_ODIN.testCases.functionalRegression" />
        </packages>
    </test> -->


    <test name="Odin_TestCases">
        <classes>
            <!--<class name="com.odin.Chitti_ODIN.testCases.regression.ForgotPasswordCases"/>




            <class name="com.odin.Chitti_ODIN.testCases.regression.BrandMasterCase"/>
            <class name="com.odin.Chitti_ODIN.testCases.regression.BrandMasterScenario"/>
            <class name="com.odin.Chitti_ODIN.testCases.regression.ContactUsCases"/>

            <class name="com.odin.Chitti_ODIN.testCases.regression.HomeValidation"/>
            <class name="com.odin.Chitti_ODIN.testCases.regression.ListingtoPDP_Validations"/>
            <class name="com.odin.Chitti_ODIN.testCases.regression.GlobalSearch"/>

            <class name="com.odin.Chitti_ODIN.testCases.functionalRegression.PDPFunctionalCases"/>

            <class name="com.odin.Chitti_ODIN.testCases.regression.ListingCases"/>
            <class name="com.odin.Chitti_ODIN.testCases.functionalRegression.miniBagCases.MiniBag_NonLoggedin"/>
            <class name="com.odin.Chitti_ODIN.testCases.regression.CategoryMasterTemplate"/>


            <class name="com.odin.Chitti_ODIN.testCases.regression.Debug"/>-->


            <!--<class name="com.odin.Chitti_ODIN.testCases.regression.CreateOrderFlows">
                <methods>
                    <include name="CreateOrderWith_NewUser" />
                    <include name="SingleItemOrder_StorePickup_COD" />
                    <include name="SingleItemOrder_StorePickup_SavedCard" />
                    <include name="createTwoItems_COD" />
            <include name="SingleItemOrder_StorePickup_NewCard" />
            <include name="createSingleOrder_NewCard" />


        </methods>
    </class>-->

            <class name="com.odin.Chitti_ODIN.testCases.regression.LoginCases"/><!---->
            <class name="com.odin.Chitti_ODIN.testCases.regression.SingupCases"/>
            <class name="com.odin.Chitti_ODIN.testCases.functionalRegression.SignUpPossibleCases"/>
            <class name="com.odin.Chitti_ODIN.testCases.functionalRegression.miniBagCases.MiniBag_LoggedIN"/>
            <class name="com.odin.Chitti_ODIN.testCases.functionalRegression.miniBagCases.MiniBag_NonLoggedin"/>
            <class name="com.odin.Chitti_ODIN.testCases.regression.MyBagCases"/>
            <class name="com.odin.Chitti_ODIN.testCases.regression.MyFavouriteCases"/>
            <class name="com.odin.Chitti_ODIN.testCases.regression.MyAccountCases"/>
            <class name="com.odin.Chitti_ODIN.testCases.regression.CreateOrderFlows"/>


            <!--<class name="com.odin.Chitti_ODIN.testCases.regression.BrandJourneyCases"/>
            -->

        </classes>


    </test>

</suite>
```

**POM FILE**
```xml
<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
	xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
	<modelVersion>4.0.0</modelVersion>
	<groupId>com.odin</groupId>
  <artifactId>Chitti-ODIN</artifactId>
  <version>0.0.3</version>
  <packaging>jar</packaging>
	<name>Chitti-ODIN</name>
	
	
	<properties>
		<project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
		<maven.compiler.source>1.8</maven.compiler.source>
		<maven.compiler.target>1.8</maven.compiler.target>
	</properties>
	<dependencies>
		<dependency>
			<groupId>junit</groupId>
			<artifactId>junit</artifactId>
			<version>3.8.1</version>
			<scope>test</scope>
		</dependency>
		<dependency>
			<groupId>org.YantriKaran</groupId>
			<artifactId>YantriKaran</artifactId>
			<version>2.9</version>
		</dependency>
		<dependency>
		<groupId>org.apache.poi</groupId>
		<artifactId>poi</artifactId>
		<version>3.13</version>
	</dependency>
	
	<dependency>
		<groupId>org.apache.poi</groupId>
		<artifactId>poi-ooxml</artifactId>
		<version>3.13</version>
	</dependency>
	
	<dependency>
    <groupId>com.googlecode.json-simple</groupId>
    <artifactId>json-simple</artifactId>
    <version>1.1.1</version>
	</dependency>
	<dependency>
		<groupId>org.apache.poi</groupId>
		<artifactId>poi-ooxml-schemas</artifactId>
		<version>3.13</version>
	</dependency>
		<dependency>
			<groupId>org.json</groupId>
			<artifactId>json</artifactId>
			<version>20160212</version>
		</dependency>
		
		 <dependency>
            <groupId>com.jayway.restassured</groupId>
            <artifactId>rest-assured</artifactId>
            <version>2.9.0</version>
            <scope>test</scope>
        </dependency>

	<dependency>
		<groupId>org.mongodb</groupId>
		<artifactId>mongo-java-driver</artifactId>
		<version>3.4.2</version>
	</dependency>
	</dependencies>
	<repositories>
		<repository>
			<id>central</id>
			<name>Central Repository</name>
			<url>http://repo.maven.apache.org/maven2</url>
			<layout>default</layout>
			<snapshots>
				<enabled>false</enabled>
			</snapshots>
		</repository>
		<repository>
			<id>ail</id>
			<name>ail releases</name>
			<releases>
				<enabled>true</enabled>
				<updatePolicy>always</updatePolicy>
				<checksumPolicy>warn</checksumPolicy>
			</releases>
			<url>http://scrolls.ailiens.com:8081/repository/maven-releases</url>
			<layout>default</layout>
		</repository>
	</repositories>

	<build>

		<plugins>
			<plugin>
				<groupId>org.apache.maven.plugins</groupId>
				<artifactId>maven-surefire-plugin</artifactId>
				<version>3.0.0-M1</version>
				<configuration>

					<suiteXmlFiles>
						<suiteXmlFile>testng.xml</suiteXmlFile>
					</suiteXmlFiles>
				</configuration>
			</plugin>
		</plugins>
	</build>
</project>
```

**Comment by kashifalam** [comment]
@juherr @krmahadevan @missedone 
Can someone help me out here!!!
Thanks in advance :)

**Comment by juherr** [comment]
@kashifalam Could you share a little project that shows the issue?

**Comment by kashifalam** [comment]
@juherr 
i cant share a project, as its very big and cant be shared with people outside my company.
ill share one of the class which is part of the testng.xml instead

Also can you give some clues as to what could be the reason for this error, so that i can at least start looking in that direction.

=================================
package com.odin.Chitti_ODIN.testCases.regression;

import com.odin.Chitti_ODIN.webScreens.*;
import common.configurations.ContextManager;
import org.testng.annotations.Test;

import com.odin.Chitti_ODIN.utils.webUtils.UserBaseTest;

import common.helpers.Log;

/**
 * @author Harikrishna Recharla SDET II
 * @date 06-Sep-2017 3:12:04 PM
 */
public class LoginCases extends UserBaseTest {

    /**
     * Submitting Valid Login Details & Verifying User
     */
    @Test(groups = {"Regress_Odin"}, priority = 1)
    public void ValidLogin() {
        Log.info("Checking Valid Login scenario");
        HomeScreen home = new HomeScreen();
        home.navigateto_Login().enterCredentials();
        home.vrfyUserHeader().signout();
    }


    /*Entering Invalid Email & pwd to Check Error Message*/
    @Test()
    public void invalidLogin() {
        Log.info("Checking InValid Login scenario");
        HomeScreen home = new HomeScreen();
        home.navigateto_Login().enterInvalidCredentials();
    }

    /**
     * Entering Empty Details & Checking Login
     */
    @Test()
    public void emptyCredentials() {
        Log.info("Checking Empty Login scenario");
        HomeScreen home = new HomeScreen();
        home.navigateto_Login().enterEmptyCredentials();
    }

    /**
     * Entering Non Registered Email to check Error message
     */
    @Test()
    public void invalidEmail() {
        Log.info("Checking Invalid Login scenario");
        HomeScreen home = new HomeScreen();
        home.navigateto_Login().enterInvalidEmail();
    }

    /**
     * Login from Lisitng -- Verify user Header
     * Verfy Listing Url
     */
    @Test
    public void loginFrom_Listing() {
        HomeScreen home = new HomeScreen();
        ListingScreen list = home.searchwith_styleid("Shirts");


        CommonElements common = new CommonElements();
        String url = common.getCurrentUrl();


        LoginScreen login = new LoginScreen();
        login.logintoApp().vrfyUserHeader();

//		String lisitngurl = common.getCurrentUrl();
//		common.assertValue(url, lisitngurl);
//		
    }


    /**
     * Login from MyBag
     */
    @Test
    public void loginFrom_MyBag() {

        HomeScreen home = new HomeScreen();
        home.searchwith_styleid("Shirts").clickOn_BrandName();

        PDPScreen pdp = new PDPScreen();
        pdp.getproductDetails();
        pdp.addtoBag();
        CommonElements ce = new CommonElements();
        ce.clickBagIcon();
        ce.naviagtetoShoppingBagFromMiniBag();
        ce.clickLoginButtonOnMybagPage();

        LoginScreen login = new LoginScreen();
        login.enterCredentials();

        //MyBagScreen mb = new MyBagScreen();
        ce.signout();

		/*CheckoutScreen check = new CheckoutScreen();
        check.signout_Checkout();*/

    }


    /**
     * Login from PDP
     * <p>
     * Search with Keyword -- Navigate to PDP
     * Add to bag -- Click on Checkout
     */
    @Test
    public void loginFrom_Checkout_via_PDP() {

        HomeScreen home = new HomeScreen();
        home.searchwith_styleid("Shirts").clickOn_BrandName();

        PDPScreen pdp = new PDPScreen();
        pdp.getproductDetails();
        pdp.addProduct_toBagClickCheckout();

        LoginScreen login = new LoginScreen();
        login.enterCredentials();

        CheckoutScreen check = new CheckoutScreen();
        check.signout_Checkout();

    }

    /**
     * Brand Joureny Login
     */
    @Test
    public void LoginFromBrandPage() {
        HomeScreen home = new HomeScreen();
        home.navigateToURL(ContextManager.getTestLevelContext().getValueAsString("BRAND_PAGE_URL"));
        home.navigateto_Login().enterCredentials();
        home.signout();
    }


}

**Comment by juherr** [comment]
@kashifalam The idea is not having the whole project, but just a minimal one. 
Most of the time, it helps users to find where is the issue in their own code ;)
Otherwise, it will help us to reproduce easily and save our time.

**Comment by krmahadevan** [comment]
Please also do the following : 

1. Retry your tests using the latest released version of TestNG (7.0.0-beta1)
2. Change the verbose attribute to "2" or higher so that we can see some of the exception information as well.

**Comment by krmahadevan** [comment]
No response from the user. Closing this issue. Please comment if this is still a problem.
